paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -202,6 +206,103 @@ def report_waiting_aborted(service, deploy_group):
     paasta_print()
 
 
+class SlackDeployNotifier(object):
+    def __init__(self, service, deploy_info, deploy_group, commit, old_commit):
+        self.sc = get_slack_client()
+        self.service = service
+        self.deploy_info = deploy_info
+        self.deploy_group = deploy_group
+        self.channels = deploy_info.get('slack_channels', [])
+        self.commit = commit
+        self.old_commit = old_commit
+
+    def deploy_group_is_set_to_notify(self):
+        for step in self.deploy_info.get('pipeline', []):
+            if step.get('step', '') == self.deploy_group:
+                return step.get('slack_notify', False)
+        return False
+
+    def post(self, **kwargs):
+        if self.deploy_group_is_set_to_notify():
+            self.sc.post(**kwargs)
```

Comment:
doesn't look like you're actually calling this method anywhere; you're calling `self.sc.post` in the methods that follow.

Reply:
whoops. Lack of testing contributed to this bug.

I've added a test and corrected it.